Slack-Apps/week3/homework/responses/3.1-reaction-event-listener-and-threads.md

```diff
@@ -0,0 +1,49 @@
+# Google Translate API: Event Listener & Threads
+
+During the livestream, you were introduced to two new concepts: a new event listener to emoji reactions and a new way of having your Slack App respond to a thread. In this issue, you will be incorportating these concepts as you integrate your Slack App with your first third party API, Google Translate!
+
+## Event Listeners
+
+Before we introduce the Google Translate API, make sure you understand how the new event listener works as well as how to make your Slack App respond to a thread:
+
+Emoji Reactions: https://api.slack.com/events/reaction_added
+ ```javascript 
+ app.event('reaction_added', async ({ event, client }) => {
+ /* Listening for Reaction Emojis */
+ });
+
+```
+Using ```app.event('reaction_added')```acts as an event listener whenever a user responds to a message with a reaction (this can be any emoji ex: smile, heart, or even a flag!). We will be using the reaction_added event listener for our Slack App that translates a user's message to a language based on the nationality of the flag the user reacts to the message with.
+
+```javascript
+const { reaction, user, item: { channel, ts } } = event;
+```
+
+Here, we are setting the reaction value to a constant in order to compare it to a list of languages. However, the names of some emojis don't exactly represent the language we need to translate to which is why we have to create a map of country codes corresponding with the APIs values. An example of this:
+```javascript
+const langMap = {
+      mx: ['es', 'Spanish'],
+      es: ['es', 'Spanish'],
+      ru: ['ru', 'Russian'],
+      jp: ['ja', 'Japanese'],
+    }
+```
+Using these two concepts, we need to create a segment of code that compares the name value of the flag to see whether the Google Translate API incorporates its value and if not, uses are map of country codes.
+
+To do this, we will need to create a new constant called ```validFlag``` and compare it with our constant called reaction (which we set already) and test if ```reaction.includes ('flag-')```. If it doesn't, we compare it to our language Map (```const langMap```). After, if ```reaction.includes ('flag-')``` we need to split the string into two parts by using ```reaction.split(-)```.
```

Comment:
Maybe add "After, if ```reaction.includes ('flag-')``` we need to split the string into two parts by using ```reaction.split(-)```." right after the ```reaction.includes ('flag-')``` portion instead of placing it last. This would more clearly show the flow of logic.